training_platform/clients/agent_client.py

```diff
@@ -105,30 +115,49 @@ def __init__(self, agent):
         self.match_maker_addr = self.asys.createActor(MatchMaker, globalName="MatchMaker")
         self.logger_addr = self.asys.createActor(Logger, globalName="Logger")
         msg = InitClientActorMsg(agent, self.match_maker_addr, self.game_manager_addr, self.logger_addr)
-        response = self.asys.ask(self.client_actor_address, msg)
-        if not isinstance(response, GameManagerInitializedMsg):
+        response = self.ask(self.client_actor_address, msg)
+        if not isinstance(response, AgentClientActorInitializedMsg):
             raise UnexpectedMessageError(response)
 
     @property
     def agent(self):
-        response = self.asys.ask(self.client_actor_address, GetAgentMsg())
+        response = self.ask(self.client_actor_address, GetAgentMsg())
         if isinstance(response, AgentMsg):
             return response.agent
-        raise UnexpectedMessageError
-
+        raise UnexpectedMessageError(response)
 
     def join(self, player):
         if self.joined:
             raise RejoiningError
-        response = self.asys.ask(self.client_actor_address, JoinMsg(player))
+        response = self.ask(self.client_actor_address, JoinMsg(player))
         if isinstance(response, MatchMakerUninitializedMsg):
             raise MatchMakerUninitializedError
         elif isinstance(response, InvalidPlayerMsg):
             raise InvalidPlayer
         elif isinstance(response, JoinAcknowledgementsMsg):
             self.joined = True
             return True
-        raise UnexpectedMessageError
+        raise UnexpectedMessageError(response)
+
+    def log(self, text, logging_level=LoggingLevel.GAME_EVENTS):
+        if self.logger_addr is not None:
+            if not LOGGING:
+                return
+            log_msg = LogMsg(text, f"AgentClientEndpoint of {self.client_actor_address} client", logging_level)
+            self.asys.tell(self.logger_addr, log_msg)
+
+    def tell(self, target_address, message):
+        self.asys.tell(target_address, message)
+        self.log(f"Sent {message} to {target_address}", LoggingLevel.PLATFORM_COMMUNICATION_MESSAGES)
+
+    def listen(self):
+        response = self.asys.listen()
+        self.log(f"Received {response}", LoggingLevel.PLATFORM_COMMUNICATION_MESSAGES)
+        return response
+
+    def ask(self, target_address, message):
+        self.tell(target_address, message)
+        return self.listen()
```

Comment:
again, po co, w sensie jak uwazasz ze ladniej jest callowac self.ask niz self.asys.ask to spoko, ale tak czy siak w tej funkcji uzylbym implementacji ''' return self.asys.ask(target_address, message) '''

Reply:
to samo